src/bp/ui-admin/src/telemetry/offlineServerFallback.ts

```diff
@@ -0,0 +1,55 @@
+import axios from 'axios'
+import _ from 'lodash'
+import ms from 'ms'
+
+import api from '../api'
+
+import { axiosConfig } from './client'
+
+const serverUrlPayloads = '/admin/telemetry-payloads'
+const serverUrlFeedback = '/admin/telemetry-feedback'
+
+const sendServerPackage = async () => {
+  let continueLoop = true
+  do {
+    try {
+      const {
+        data: { events }
+      } = await api.getSecured().get(serverUrlPayloads)
+
+      if (_.isEmpty(events)) {
+        break
+      }
+
+      const status = await postStorageServer(events)
+      continueLoop = status === 'ok'
+
+      await api.getSecured().post(serverUrlFeedback, { events: events.map(e => e.uuid), status })
+    } catch (err) {
+      console.error('Could not access the botpress server', err)
+
+      continueLoop = false
+    }
+  } while (continueLoop)
+}
```

Comment:
This still a while loop!! Never ever use while(true) loops. 

Why don't you just callyour sendServerPackage method after you have finished processing a batch ? If there's no more items to process, it'll exit gracefully.

Reply:
Fixed in : 
1a22377